src/Controller/SearchController.php

```diff
@@ -0,0 +1,145 @@
+        SearchEngineInterface $searchEngine,
+        FormFactoryInterface $formFactory
+    ) {
+        $this->restViewHandler = $restViewHandler;
+        $this->searchEngine = $searchEngine;
+        $this->formFactory = $formFactory;
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function filterAction(Request $request)
+    {
+        $view = View::create();
+        if ($this->isHtmlRequest($request)) {
+            $view->setTemplate($this->getTemplateFromRequest($request));
+        }
+
+        $form = $this->formFactory->create(FilterSetType::class, Criteria::fromQueryParameters($this->getResourceClassFromRequest($request), []), ['filter_set' => $this->getFilterScopeFromRequest($request)]);
+        $form->handleRequest($request);
+        $criteria = $form->getData();
+        $criteria = Criteria::fromQueryParameters($this->getResourceClassFromRequest($request), array_merge($request->query->all(), $request->attributes->all(), $criteria->getFiltering()->getFields()));
+
+        $result = $this->searchEngine->match($criteria);
+        $partialResult = $result->getResults($criteria->getPaginating()->getOffset(), $criteria->getPaginating()->getItemsPerPage());
+
+        $view->setData([
+            'resources' => $partialResult->toArray(),
+            'form' => $form->createView(),
+            'criteria' => $criteria,
+        ]);
+
+        return $this->restViewHandler->handle($view);
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return string
+     */
+    private function getTemplateFromRequest(Request $request)
+    {
+        $syliusAttributes = $request->attributes->get('_sylius');
+
+        if (!isset($syliusAttributes['template'])) {
+            throw new HttpException(Response::HTTP_BAD_REQUEST, 'You need to configure template in routing!');
+        }
+
+        return $syliusAttributes['template'];
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return string
+     */
+    private function getResourceClassFromRequest(Request $request)
+    {
+        $syliusAttributes = $request->attributes->get('_sylius');
+
+        if (!isset($syliusAttributes['resource_class'])) {
+            throw new HttpException(Response::HTTP_BAD_REQUEST, 'You need to configure resource class in routing!');
+        }
+
+        return $syliusAttributes['resource_class'];
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return bool
+     */
+    private function isHtmlRequest(Request $request)
+    {
+        return !$request->isXmlHttpRequest() && 'html' === $request->getRequestFormat();
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return string
+     */
+    private function getFilterScopeFromRequest(Request $request)
+    {
+        $syliusAttributes = $request->attributes->get('_sylius');
+
+        if (!isset($syliusAttributes['filter_set'])) {
```

Comment:
This means I will need to use a different route for each filter set. Fine for now, but we should get it from the Taxon. ;) This way we can configure it per category.